### DISCUSSION TITLE
How to use Chakra UI v3 in shadow dom with React 19

### DISCUSSION THREAD

**Discussion by lhk** [chakra-ui/chakra-ui#9338] [Q&A] [answered]
### Description

I'd like to display some part of my UI in a shadow DOM. There's a section in the docs for this purpose: https://www.chakra-ui.com/docs/get-started/environments/shadow-dom

They rely on the `react-shadow` package which is not compatible with React 19.

### Link to Reproduction

https://github.com/chakra-ui/chakra-ui/tree/main/sandbox/shadow-dom

### Chakra UI Version

3.0.0

### Operating System

- [x] macOS
- [x] Windows
- [x] Linux

**Comment by segunadebayo** [comment]
I'd recommend opening an issue with react-shadow to support React 19, if you don't have time to implement yours.

**Comment by segunadebayo** [comment, marked as answer]
The docs path still holds. `react-shadow` is what is missing React 19.

Open that with them, or mount your own shadow root and pass it to `EnvironmentProvider`. https://chakra-ui.com/docs/get-started/environments/shadow-dom
